wear/src/main/java/io/homeassistant/companion/android/home/views/SettingsView.kt

```diff
@@ -1,66 +1,175 @@
-                .fillMaxWidth(),
-            icon = {
-                Image(asset = CommunityMaterial.Icon3.cmd_star)
-            },
-            label = {
-                Text(
-                    text = stringResource(id = R.string.favorite)
+    val scalingLazyListState: ScalingLazyListState = rememberScalingLazyListState()
+    RotaryEventState(scrollState = scalingLazyListState)
+    ScalingLazyColumn(
+        modifier = Modifier
+            .fillMaxSize(),
+        contentPadding = PaddingValues(
+            top = 10.dp,
+            start = 5.dp,
+            end = 5.dp,
+            bottom = 40.dp
+        ),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        state = scalingLazyListState
+    ) {
+        item {
+            ListHeader(id = R.string.settings)
+        }
+
+        item {
+            Chip(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(top = 20.dp),
+                icon = {
+                    Image(asset = CommunityMaterial.Icon3.cmd_star)
+                },
+                label = {
+                    Text(
+                        text = stringResource(id = R.string.favorite)
+                    )
+                },
+                onClick = onClickSetFavorites,
+                colors = ChipDefaults.primaryChipColors(
+                    contentColor = Color.Black
                 )
-            },
-            onClick = onClickSetFavorites,
-            colors = ChipDefaults.primaryChipColors(
-                contentColor = Color.Black
             )
-        )
-        Chip(
-            modifier = Modifier
-                .fillMaxWidth()
-                .padding(top = 10.dp),
-            icon = {
-                Image(asset = CommunityMaterial.Icon.cmd_delete)
-            },
-            label = {
-                Text(
-                    text = stringResource(id = R.string.clear_favorites),
+        }
+
+        item {
+            Chip(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(top = 10.dp),
+                icon = {
+                    Image(asset = CommunityMaterial.Icon.cmd_delete)
+                },
+                label = {
+                    Text(
+                        text = stringResource(id = R.string.clear_favorites),
+                    )
+                },
+                onClick = onClearFavorites,
+                colors = ChipDefaults.primaryChipColors(
+                    contentColor = Color.Black
+                ),
+                secondaryLabel = {
+                    Text(
+                        text = stringResource(id = R.string.irreverisble)
+                    )
+                },
+                enabled = favorites.isNotEmpty()
+            )
+        }
+
+        item {
+            val haptic = LocalHapticFeedback.current
+            var checked by rememberSaveable { mutableStateOf(isHapticEnabled) }
```

Comment:
We don't want to remember between recompositions, we should always use `isHapticEnabled`

Reply:
so without this and even with using the viewmodel variable the checked status does not update until the page is recomposed which only happens when I navigate to another screen and back